src/coreclr/nativeaot/Runtime/amd64/ExceptionHandling.asm

```diff
@@ -642,6 +644,9 @@ endif
 
 ALTERNATE_ENTRY RhpCallFinallyFunclet2
 
+        vzeroupper
+
+if 0 ;; Any register value changes made in the funclet are lost in the current ABI
```

Comment:
Was this just a testing change that should be removed? Your comment says that we are unifying the coreclr and nativeAOT behavior and this is going the other way.